### PR TITLE
NHSNLINK-386 - Validation of access token used in REST requests

### PR DESCRIPTION
**Pull request by arianamihailescu**

### CODE REVIEW

File: web/src/main/angular/app/services/auth.service.ts [hunk 1]
```diff
@@ -64,11 +64,8 @@ export class AuthService {
       if (!this.user) {
         this.oauthService.initImplicitFlow(encodeURIComponent(this.router.url));
       } else {
-        console.log('Your token is: ' + this.token);
-        await this.oauthService.setupAutomaticSilentRefresh();
-
         this.token = this.oauthService.getIdToken();
-
+        console.log('Your token is: ' + this.token);
```
Comment (seanmcilvenna) [inline web/src/main/angular/app/services/auth.service.ts:68]: Remove this console.log please

File: web/src/main/angular/app/services/auth.service.ts [hunk 2]
```diff
@@ -79,10 +76,17 @@ export class AuthService {
         if (path && path !== '/') {
           this.router.navigate([path]);
         }
+        await this.oauthService.setupAutomaticSilentRefresh();
```
Comment (seanmcilvenna) [inline web/src/main/angular/app/services/auth.service.ts:79]: Did this get removed accidentally somewhere? Thought we had this already...

**Review by seanmcilvenna — CHANGES_REQUESTED** [review]

File: web/src/main/angular/app/services/auth.service.ts (hunk 1 above)
Comment (arianamihailescu) [reply web/src/main/angular/app/services/auth.service.ts]: Removed

File: web/src/main/angular/app/services/auth.service.ts (hunk 2 above)
Comment (arianamihailescu) [reply web/src/main/angular/app/services/auth.service.ts]: Yes - but needed some changes around it to be sure we have the latest token.

**Pull request merged**